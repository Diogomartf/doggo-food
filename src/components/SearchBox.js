import React from 'react';
import styled from 'styled-components';
import { Search } from 'react-feather';

const SearchBox = ({
  handleSubmit,
  handleChange,
  searchText,
  handleOnFocus,
  handleOnBlur
}) => (
  <SearchBoxStyled>
    <SearchInputForm onSubmit={handleSubmit}>
      <SearchIconStyled size={18} />
      <InputStyled
        type="text"
        placeholder="Search Recipes.."
        value={searchText}
        onChange={e => handleChange(e.target.value)}
        onFocus={handleOnFocus}
        // onBlur={handleOnBlur}
      />
    </SearchInputForm>
  </SearchBoxStyled>
);

const SearchBoxStyled = styled.div`
  padding: 12px 24px;
  background: ${({ theme }) => theme.colors.lightGray};
`;

const SearchInputForm = styled.form`
  display: flex;
  align-items: center;
  height: 47px;
  padding: 0 12px;
  border-radius: 8px;
  background: ${({ theme }) => theme.colors.gray};
`;

SearchInputForm.displayName = 'SearchInputForm';

const SearchIconStyled = styled(Search)`
  margin-right: 12px;
  color: ${({ theme }) => theme.colors.darkGray};
`;

const InputStyled = styled.input`
  width: 100%;
  border: none;
  background: none;
  font-weight: 600;
  font-size: 14px;
  color: ${({ theme }) => theme.colors.darkGray};

  :focus {
    outline: none;
  }
`;

export default SearchBox;
