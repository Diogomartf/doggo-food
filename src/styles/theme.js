import colors from './colors';

export default {
  name: 'default theme',
  colors
};
