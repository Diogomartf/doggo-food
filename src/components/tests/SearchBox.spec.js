import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../../styles';

import SearchBox from '../SearchBox';

describe('SearchBox', () => {
  it('should render correctly', () => {
    const handleSubmitMock = jest.fn();
    const handleChangeMock = jest.fn();

    const wrapper = shallow(
      <SearchBox
        handleSubmit={handleSubmitMock}
        handleChange={handleChangeMock}
        searchText={''}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('handleSubmit is called with correct data', async () => {
    const handleSubmitMock = jest.fn();
    const handleChangeMock = jest.fn();
    const data = { somedata: 'data' };
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <SearchBox
          handleSubmit={handleSubmitMock}
          handleChange={handleChangeMock}
          searchText={''}
        />
      </ThemeProvider>
    );

    wrapper
      .find('SearchInputForm')
      .props()
      .onSubmit(data);
    expect(handleSubmitMock).toBeCalledWith({ somedata: 'data' });
  });
});
