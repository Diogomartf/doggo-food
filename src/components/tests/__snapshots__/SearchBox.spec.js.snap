// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render correctly 1`] = `
<styled.div>
  <SearchInputForm
    onSubmit={[MockFunction]}
  >
    <Styled(Search)
      size={18}
    />
    <styled.input
      onChange={[MockFunction]}
      placeholder="Search Recipes.."
      type="text"
      value=""
    />
  </SearchInputForm>
</styled.div>
`;
